src/MigrationTools.Host/Services/DetectVersionService.cs

```diff
@@ -28,40 +33,28 @@ public Version GetLatestVersion()
             Stopwatch mainTimer = Stopwatch.StartNew();
             //////////////////////////////////
             Version latestPackageVersion = null;
-            string packageID = "vsts-sync-migrator";
             bool sucess = false;
             try
             {
-                //Connect to the official package repository
-                IEnumerable<NuGetVersion> versions = GetChocoVersions(packageID);
-                latestPackageVersion = versions.Max(p => p.Version);
+                WinGetPackageManager packageManager = new WinGetPackageManager();
+                var package = packageManager.SearchPackage(PackageId).SingleOrDefault();
```

Comment:
The package search will search the winget repository and will always return the newest version. Package.Version and Package.AvailableVersion will always be the same here.

You should use `GetInstalledPackages(packageId)` instead of “SearchPackage(PackageId)”. This will give you the correct installed and available version and should also be faster, because you are just going through the installed packages.

Reply:
I created a pull request to fix this problem. (#1684)

Reply:
I do want the latest available version from winget and not the locally installed version.

I believe that's Search and not Installed.

Reply:
The available version should always be up to date with the one provided from the repository, even if you just list the installed packages.
At least that is what I understood from reading the documentation: https://learn.microsoft.com/en-us/windows/package-manager/winget/list#list-with-update

When using `GetInstalledPackages(id)` you will get the installed and the available version in the returned object.

When using the search you will get no information about the installed version.  
You are currently comparing it to the assembly version, so it should still work the way it is now.
It might just come down to preference or what else the package object will be used for, because the package state ist not completely “correct” if it is created from the search.

Reply:
There might also be a risk of there being multiple entries in the search with a similar ID.
The best match should be on top, but I’m not sure if I would always trust that.

```
Name              Id                  Version    Source
--------------------------------------------------------
7-Zip             7zip.7zip           23.01      winget
7-Zip Alpha (msi) 7zip.7zip.Alpha.msi 21.03.00.0 winget
7-Zip Alpha (exe) 7zip.7zip.Alpha.exe 23.00      winget
```

This could be worked around by using the `--exact` flag, but it is currently not supported in the wrapper. But I might add it soon to fix this.